### PR TITLE
Use PyInt::try_to_primitive.

### PR DESCRIPTION
**Pull request by DimitrisJim**
clean up after #3142

### CODE REVIEW

File: vm/src/iterator.rs [hunk 1]
```diff
@@ -66,20 +65,19 @@ pub fn length_hint(vm: &VirtualMachine, iter: PyObjectRef) -> PyResult<Option<us
...
                 result.class().name()
             ))
         })?
-        .as_bigint();
-    if result.is_negative() {
-        return Err(vm.new_value_error("__length_hint__() should return >= 0".to_owned()));
+        .try_to_primitive(vm)
+    {
+        Ok(hint) => Ok(Some(hint)),
+        Err(_) => Err(vm.new_value_error("__length_hint__() should return >= 0".to_owned())),
```
Comment (youknowone) [inline vm/src/iterator.rs:79]: The original intend looks like distinguising overflow error and negative value error

**Comment by youknowone** [comment]
An idea. Sometimes, there will be some features we cannot move from BigInt to PyInt. Would it be better if I didn't move try_to_primitive into PyInt but adding `trait PyIntExt` and impl it for BigInt?

File: vm/src/iterator.rs (hunk 1 above)
Comment (DimitrisJim) [reply vm/src/iterator.rs:79]: ah, good point. I'll address that.

**Comment by DimitrisJim** [comment]
@youknowone are there places that would need that now? `try_from_primitive` was the last (as far as I can see) in `int.rs` that was a free-standing public function that accepted a `BigInt`.

**Review by youknowone — APPROVED** [review]

**Comment by youknowone** [comment]
Probably `try_f64_to_bigint` and `try_bigint_to_f64` are one of those example for same relationship. And then I also think try_from_primitive is also one of the member of the group

**Pull request merged**